sql/service_wsrep.cc

```diff
@@ -186,42 +186,28 @@ extern "C" my_bool wsrep_thd_is_SR(const THD *thd)
   return thd && thd->wsrep_cs().transaction().is_streaming();
 }
 
-extern "C" void wsrep_handle_SR_rollback(THD *bf_thd,
+extern "C" void wsrep_handle_SR_rollback(THD *bf_thd __attribute__((unused)),
                                          THD *victim_thd)
 {
+  /*
+    We should always be in victim_thd context, either client session is
+    rolling back or rollbacker thread should be in control.
+  */
   DBUG_ASSERT(victim_thd);
-  DBUG_ASSERT(wsrep_thd_is_SR(victim_thd));
```

Comment:
I think we could keep `DBUG_ASSERT(wsrep_thd_is_SR(victim_thd))`.
Is there a reason to remove it?

Reply:
Done